### PR TITLE
Configurable Protocol Magic

### PR DESCRIPTION
**Pull request by karknu**
Breaks backwards compatibility.
Depends on https://github.com/input-output-hk/ouroboros-network/pull/1142/

### CODE REVIEW

File: cardano-node/src/Cardano/CLI/Tx.hs [hunk 1]
```diff
@@ -144,6 +144,15 @@ withRealPBFT co cc action = do
     proto@Consensus.ProtocolRealPBFT{} -> action proto
     _ -> throwIO $ ProtocolNotSupported (coProtocol co)
 
+-- | Return an id of GenTx.
+byronGenTxId
```
Comment (mrBliss) [inline cardano-node/src/Cardano/CLI/Tx.hs]: @denisshevchenko This function exists already, it is called `txId`

Comment (karknu) [reply cardano-node/src/Cardano/CLI/Tx.hs]: I've modified the code to use `txId`.

File: cardano-node/src/Cardano/CLI/Tx.hs [hunk 2]
```diff
@@ -43,10 +43,10 @@ import qualified Cardano.Crypto.Hashing as Crypto
 import qualified Cardano.Crypto.Signing as Crypto
 import qualified Ouroboros.Consensus.Demo.Run as Demo
 import qualified Ouroboros.Consensus.Ledger.Byron as Byron
-import           Ouroboros.Consensus.Ledger.Byron ( GenTx(..), GenTxId(..)
-                                                  , ByronBlockOrEBB)
+import           Ouroboros.Consensus.Ledger.Byron ( GenTx(..),ByronBlockOrEBB)
```
Comment (mrBliss) [inline cardano-node/src/Cardano/CLI/Tx.hs]:
```suggestion
import           Ouroboros.Consensus.Ledger.Byron (GenTx(..), ByronBlockOrEBB)
```

**Comment by karknu** [comment]
This PR MUST be merged together with updates in `cardano-byron-broxy` and `cardano-explorer` since it breaks backwards compatibility.

**Comment by mrBliss** [comment]
> This PR MUST be merged together with updates in `cardano-byron-broxy` and `cardano-explorer` since it breaks backwards compatibility.

Is that true for the `cardano-byron-proxy`? It does not depend on `cardano-node` (but on `ouroboros-network`).

**Comment by karknu** [comment]
If `cardano-byron-proxy` isn't updated at the same time `cardano-node` would not be able to sync with it which would make for unhappy testers and developers who depend on it.

**Comment by mrBliss** [comment]
> If `cardano-byron-proxy` isn't updated at the same time `cardano-node` would not be able to sync with it which would make for unhappy testers and developers who depend on it.

Ah, indeed, good point :+1:

File: cardano-node/src/Cardano/Node/Run.hs [hunk 3]
```diff
@@ -182,18 +182,19 @@ handleSimpleNode p trace nodeTracers cc = do
...
+          , rnaDnsResolverTracer       = dnsResolverTracer     nodeTracers
+          , rnaMuxTracer               = muxTracer             nodeTracers
+          , rnaMuxLocalTracer          = nullTracer
+          , rnaMkPeer                  = Peer
+          , rnaMyAddrs                 = addrs
+          , rnaMyLocalAddr             = myLocalAddr
+          , rnaIpProducers             = ipProducers
+          , rnaDnsProducers            = dnsProducers
+          , rnaHandshakeTracer         = nullTracer
+          , rnaHandshakeLocalTracer    = nullTracer
```
Comment (mrBliss) [inline cardano-node/src/Cardano/Node/Run.hs]: There are some extra unnecessary spaces before the `=`. These changes can be undone, only the line below needs to be added.

**Pull request merged**